src/traceViewer.ts

```diff
@@ -19,9 +19,20 @@ import type { TestConfig } from './playwrightTestTypes';
 import { findNode } from './utils';
 import * as vscodeTypes from './vscodeTypes';
 
-export type TraceViewer = SpawnTraceViewer;
+export type TraceViewer = {
+  currentFile(): string | undefined;
+  willRunTests(): Promise<void>;
+  open(file?: string): Promise<void>;
+  close(): void;
+  infoForTest(): {
+    type: string;
+    serverUrlPrefix?: string;
+    testConfigFile: string;
+    traceFile?: string;
+  } | undefined;
+};
 
-export class SpawnTraceViewer {
+export class SpawnTraceViewer implements TraceViewer {
```

Comment:
Move this out into its own file as a separate change and then you can place TraceViewerFactory here.

Reply:
Addressed in 87543434f44c7494c7e7e46678f88bd4173db83a